tasks/rubies.yml

```diff
@@ -62,16 +62,24 @@
   when: not '--user-install' in rvm1_install_flags and rvm1_bundler_install
   with_items: '{{ rvm1_symlink_bundler_binaries }}'
 
-- name: List installed versions of Ruby (for deletion)
+- name: List installed ruby versions (for deletion)
   command: '{{ rvm1_rvm }} list strings'
   changed_when: False
   register: rvm1_list_installed_rubies
-  when: rvm1_delete_ruby is defined and rvm1_delete_ruby != ''
+  when: rvm1_delete_unmanaged_rubies|bool or (rvm1_delete_ruby is defined and rvm1_delete_ruby != '')
 
-- name: Delete ruby if relevant
+- name: Delete unmanaged ruby versions
+  command: '{{ rvm1_rvm }} remove {{ item }}'
+  register: rvm_delete_rubies_command_result
+  changed_when: "'#removing' in rvm_delete_rubies_command_result.stdout"
```

Comment:
Not sure if we need the `changed_when` here since at this point change is always expected for the given list.

Reply:
Thank you, quite correct. I did some copy-paste without giving it more thoughts.

Reply:
Fixed in fada459a67fc67e0d5a11dee507d2ab76ecc7a3a